### PR TITLE
[Fix-13434][Resource Center] fix the NPE when use OSS as the resource center

### LINKED ISSUE #13434: [Bug] [Resource Center] NPE when use OSS as the resource center
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

* NPE when use OSS as the resource center
截屏2023-01-20 10 18 21

image

### What you expected to happen

no NPE

### How to reproduce

use OSS as the resource center

### Anything else

* After the resource center is refactored in #12076
* The new methods in `StorageOperate` is not implemented in `OSS`
image

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by rickchengx**
## Purpose of the pull request

* close: #13434 

* After the resource center is refactored in #12076
* The new methods in `StorageOperate` is not implemented in `OSS`
image

## Verify this pull request

manually tested

截屏2023-01-20 11 44 52

截屏2023-01-20 11 44 59

### CODE REVIEW

**Comment by rickchengx** [comment]
Hi, @EricPyZhou @EricGao888 @SbloodyS  Could you please help review this?

File: dolphinscheduler-storage-plugin/dolphinscheduler-storage-oss/src/main/java/org/apache/dolphinscheduler/plugin/storage/oss/OssStorageOperator.java [hunk 1]
```diff
@@ -288,19 +309,171 @@ public ResUploadType returnStorageType() {
     @Override
     public List<StorageEntity> listFilesStatusRecursively(String path, String defaultPath, String tenantCode,
                                                           ResourceType type) {
-        return null;
+        List<StorageEntity> storageEntityList = new ArrayList<>();
+
+        LinkedList<StorageEntity> foldersToFetch = new LinkedList<>();
```
Comment (ruanwenjun) [inline dolphinscheduler-storage-plugin/dolphinscheduler-storage-oss/src/main/java/org/apache/dolphinscheduler/plugin/storage/oss/OssStorageOperator.java:313]: It's better to add initialize path in `foldersToFetch`, and then you only need to loop the `foldersToFetch`.

```java
while (!foldersToFetch.isEmpty()) {
  String pathToExplore = foldersToFetch.pop().getFullName();
   try {
           List<StorageEntity> tempList = listFilesStatus(pathToExplore, defaultPath, tenantCode, type);
           for (StorageEntity temp : tempList) {
               if (temp.isDirectory()) {
                   foldersToFetch.add(temp);
               }
           }
           storageEntityList.addAll(tempList);
       } catch (Exception e) {
           logger.error("error while listing files status recursively, path: {}", pathToExplore, e);
       }
}


```

**Review by ruanwenjun — COMMENTED** [review]
Basically LGTM

**Comment by ruanwenjun** [comment]
Since we don't hope to add new feature in 3.1.x, so I add lable 3.2.0.

**Review by EricGao888 — COMMENTED** [review]
Good job for fixing these methods. Could u plz also add some related UT cases?

File: dolphinscheduler-storage-plugin/dolphinscheduler-storage-oss/src/main/java/org/apache/dolphinscheduler/plugin/storage/oss/OssStorageOperator.java (hunk 1 above)
Comment (rickchengx) [reply dolphinscheduler-storage-plugin/dolphinscheduler-storage-oss/src/main/java/org/apache/dolphinscheduler/plugin/storage/oss/OssStorageOperator.java:313]: Hi, @ruanwenjun , thanks for your suggestion!

I've changed as below, (also in S3)

image

**Comment by rickchengx** [comment]
> Good job for fixing these methods. Could u plz also add some related UT cases?
Hi @EricGao888 , thanks for your suggestion and I've added related UTs.

**Review by EricGao888 — APPROVED** [review]

**Review by Amy0104 — APPROVED** [review]
The front end part LGTM.

**Pull request merged**